### PR TITLE
Securonix enhancements  - create-incident/watchlist

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/21227

## Description
- Added the **Host** parameter, to allow overriding the default hostname.
  - Added 4 commands.
    - ***securonix-create-incident***
    - ***securonix-create-watchlist***
    - ***securonix-check-entity-in-watchlist***
    - ***securonix-add-entity-to-watchlist***

## Minimum version of Demisto
- [x] 4.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests - unit test for each added command
- [ ] Documentation - soon

## Change to default behaviour 
action argument in the securonix-perform-action-on-incident
was changed from drop down to free text,
as securonix users can create their own workflows, with their unique actions.
I / @Arsenikr Can explain in person.

- [ ] [RN](https://github.com/demisto/content/blob/7346df118b030302042782e3caf3b33864e7d714/Integrations/Securonix/CHANGELOG.md)
- [ ] [YML](https://github.com/demisto/content/blob/7346df118b030302042782e3caf3b33864e7d714/Integrations/Securonix/Securonix.yml)
- [ ] [README](https://github.com/demisto/content/blob/7346df118b030302042782e3caf3b33864e7d714/Integrations/Securonix/README.md)
@Arsenikr FYI

### CODE REVIEW

**Comment by bakatzir** [comment]
I'll generate new and full README after tech writer review.

File: Integrations/Securonix/Securonix.py [hunk 1]
```diff
@@ -838,6 +932,47 @@ def add_comment_to_incident(client: Client, args: Dict) -> Tuple[str, Dict, Dict
...
+    if 'create incident' in action_name and not workflow:
+        raise Exception(f'Creating an incident with the action: {action_name}, Supply a workflow.')
+    response = client.create_incident_request(violation_name, resource_group, resource_name, entity_type, entity_name,
+                                              action_name, workflow, comment, criticality)
+    result = response.get('result')
+    message = response.get('messages')
+    if not result or not message or (isinstance(message, list) and 'Invalid' in message[0]):
+        if isinstance(message, list):
+            message = message[0]
+        raise Exception(f'Failed to create the incident with message:\n{message}')
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: handle case where message == [].
this is relevant for 2 of the if statement. consider changing both.

File: Integrations/Securonix/Securonix.py [hunk 2]
```diff
@@ -888,6 +1023,76 @@ def get_watchlist(client: Client, args) -> Tuple[str, Dict, Dict]:
...
+        args: Usually demisto.args()
+    Returns:
+        Outputs.
+    """
+    watchlist_name = args.get('watchlist_name')
+
+    response = client.create_watchlist_request(watchlist_name)
+
+    if 'successfully' not in response:
+        raise Exception(f'Failed to list watchlists.')
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: maybe add response in the Exception? or is there a reason? need to see how dose it look.

File: Integrations/Securonix/Securonix.py [hunk 3]
```diff
@@ -888,6 +1023,76 @@ def get_watchlist(client: Client, args) -> Tuple[str, Dict, Dict]:
...
+    """
+    watchlist_name = args.get('watchlist_name')
+    entity_type = args.get('entity_type')
+    entity_name = args.get('entity_name')
+    expiry_days = args.get('expiry_days') if 'expiry_days' in args else '30'
+
+    response = client.add_entity_to_watchlist_request(watchlist_name, entity_type, entity_name, expiry_days)
+
+    if 'successfull' not in response:
+        raise Exception(f'Failed to add entity {entity_name} to the watchlist {watchlist_name}.\n'
```
Comment (yuvalbenshalom) [inline Integrations/Securonix/Securonix.py]: maybe add response in the Exception? or is there a reason? need to see how dose it look.

**Comment by bakatzir** [comment]
@yuvalbenshalom Tended to all the error messages:
Screen Shot 2020-01-29 at 10 25 24

**Comment by kirbles19** [comment]
@bakatzir Done.

**Pull request merged**